internal/controller/storageclient_controller.go

```diff
@@ -309,26 +416,125 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
+				snapshotClass.Name = eResource.Name
+				removeStorageClaimAsOwner(snapshotClass)
+				if err := r.own(snapshotClass); err != nil {
+					return fmt.Errorf("failed to own VolumeSnapshotClass resource: %v", err)
+				}
 
-	if utils.AddAnnotation(r.storageClient, utils.DesiredConfigHashAnnotationKey, storageClientResponse.DesiredConfigHash) {
-		if err := r.update(r.storageClient); err != nil {
-			return reconcile.Result{}, fmt.Errorf("failed to update StorageClient with desired config hash annotation: %v", err)
+				if snapshotClass.Parameters == nil {
+					snapshotClass.Parameters = map[string]string{}
+				}
+				snapshotClass.Parameters["csi.storage.k8s.io/snapshotter-secret-namespace"] = r.OperatorNamespace
+
+				utils.AddLabels(snapshotClass, eResource.Labels)
+				utils.AddAnnotations(snapshotClass, eResource.Annotations)
+				return nil
+			}); err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to create or update VolumeSnapshotClass: %s", err)
+			}
+		case "VolumeGroupSnapshotClass":
+			// check for CRD availability
+			if r.AvailableCrds[VolumeGroupSnapshotClassCrdName] {
+				volumeGroupSnapshotClass := &groupsnapapi.VolumeGroupSnapshotClass{}
+				volumeGroupSnapshotClass.Name = eResource.Name
+				err = utils.CreateOrReplace(r.ctx, r.Client, volumeGroupSnapshotClass, func() error {
+					err = json.Unmarshal(eResource.Data, &volumeGroupSnapshotClass)
+					if err != nil {
+						return fmt.Errorf("failed to unmarshal VolumeGroupSnapshotClass configuration response: %v", err)
+					}
+					volumeGroupSnapshotClass.Name = eResource.Name
+					removeStorageClaimAsOwner(volumeGroupSnapshotClass)
+					if err := r.own(volumeGroupSnapshotClass); err != nil {
+						return fmt.Errorf("failed to own VolumeGroupSnapshotClass resource: %v", err)
+					}
+
+					if volumeGroupSnapshotClass.Parameters == nil {
+						volumeGroupSnapshotClass.Parameters = map[string]string{}
+					}
+					volumeGroupSnapshotClass.Parameters["csi.storage.k8s.io/group-snapshotter-secret-namespace"] = r.OperatorNamespace
+
+					utils.AddLabels(volumeGroupSnapshotClass, eResource.Labels)
+					utils.AddAnnotations(volumeGroupSnapshotClass, eResource.Annotations)
+					return nil
+				})
+				if err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to create or update VolumeGroupSnapshotClass: %s", err)
+				}
+			}
+		case "VolumeReplicationClass":
+			vrc := &replicationv1a1.VolumeReplicationClass{}
+			vrc.Name = eResource.Name
+			err := utils.CreateOrReplace(r.ctx, r.Client, vrc, func() error {
+				removeStorageClaimAsOwner(vrc)
+				if err := r.own(vrc); err != nil {
+					return fmt.Errorf("failed to own VolumeReplicationClass resource: %v", err)
+				}
+				if err := json.Unmarshal(eResource.Data, &vrc.Spec); err != nil {
+					return fmt.Errorf("failed to unmarshall VolumeReplicationClass spec: %v", err)
+				}
+				vrc.Spec.Parameters["replication.storage.openshift.io/replication-secret-namespace"] = r.OperatorNamespace
```

Comment:
Why are we adding these parameters here and not on the provider?

Reply:
#323 is the client side fix and provider side will be done at a later point.